tests/fsharpqa/Source/nunitConf.fs

```diff
@@ -0,0 +1,420 @@
+                log "Errors:"
+                err |> Array.iter (fun (l,e) -> log "- %s. Line '%s'" e l)
+                failwithf "Invalid env.lst file '%s'. Errors: %A" path err
+
+            match lines |> Array.choose (function l, Choice1Of2(Some(EnvLstLine.Data(d))) -> Some(l,d) | _ -> None) with
+            | [| |] ->
+                failwithf "No valid lines inside env.lst file '%s'" path
+            | validLines ->
+                validLines
+                |> Array.map (fun (line,data) ->
+                    let tc = EnvLstLineTestCaseData (dir, line)
+                    
+                    //the id of test, it's the comment string
+                    let testName =
+                        data.Comment
+                        |> Option.map (fun s -> s.Trim())
+                        |> Option.map (fun s -> s.Replace(".", "_")) //test name is a full class name, the dot mean nesting
+                    
+                    testName
+                    |> Option.iter (fun testName -> tc.SetName(testName) |> ignore)
+
+                    _builder.BuildTestMethod(methodInfo, suite, tc) )
+                |> Seq.ofArray
+
+
+module FSharpQATestSuite =
+
+    let envLstData () = attempt {
+        let test = NUnit.Framework.TestContext.CurrentContext.Test
+        let line = test.Properties.Get("ENVLSTLINE") :?> string
+        
+        return! 
+            match line |> EnvLst.parseLine with
+            | Choice1Of2 (Some(Data(a))) -> succeed a.Vars
+            | Choice1Of2 (Some(Comment(c))) -> genericError (sprintf "The env.lst line '%s' is a comment" line)
+            | Choice1Of2 None -> genericError (sprintf "Invalid env.lst line '%s'" line)
+            | Choice2Of2 f -> genericError (sprintf "Invalid env.lst line '%s': %A" line f)
+
+        }
+
+module FileGuard =
+    let private remove path = if File.Exists(path) then Commands.rm (Path.GetTempPath()) path
+
+    [<AllowNullLiteral>]
+    type T (path: string) =
+        member x.Path = path
+        interface IDisposable with
+            member x.Dispose () = remove path
+
+    let create path =
+        if not (Path.IsPathRooted(path)) then failwithf "path '%s' must be absolute" path
+        remove path
+        new T(path)
+    
+    let exists (guard: T) = guard.Path |> File.Exists
+        
+
+let checkGuardExists guard = processor {
+    if not <| (guard |> FileGuard.exists)
+    then return! genericError (sprintf "exit code 0 but %s file doesn't exists" (guard.Path |> Path.GetFileName))
+    }
+
+
+let check (f: Attempt<_,_>) =
+    f |> Attempt.Run |> checkTestResult
+
+
+type RedirectInfo = 
+    { Output : RedirectTo
+      Input : RedirectFrom option }
+
+and RedirectTo = 
+    | Inherit
+    | Output of RedirectToType
+    | OutputAndError of RedirectToType
+    | Error of RedirectToType
+
+and RedirectToType = 
+    | Overwrite of FilePath
+    | Append of FilePath
+
+and RedirectFrom = 
+    | RedirectInput of FilePath
+
+
+module Command =
```

Comment:
yes, this code was in `nunitConf.fs` (the only file i cannot share between suites), because i didnt refactor enough. i'll refactor and move it to a better place, shared with fsharp suite